src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -99,6 +101,8 @@ void
 mongoc_auto_encryption_opts_set_keyvault_client_pool (
    mongoc_auto_encryption_opts_t *opts, mongoc_client_pool_t *pool)
 {
+   BSON_ASSERT_PARAM (pool);
```

Comment:
```suggestion
```

Suggest removing this assertion. A caller may be relying on passing `NULL` to clear the option.